bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeperAdmin.java

```diff
@@ -535,89 +549,26 @@ public void recoverComplete(int rc, Object ctx) {
      * @param bookieSrc
      *            Source bookie that had a failure. We want to replicate the
      *            ledger fragments that were stored there.
-     * @param bookieDest
-     *            Optional destination bookie that if passed, we will copy all
-     *            of the ledger fragments from the source bookie over to it.
      * @param cb
      *            RecoverCallback to invoke once all of the data on the dead
      *            bookie has been recovered and replicated.
      * @param context
      *            Context for the RecoverCallback to call.
      */
-    public void asyncRecoverBookieData(final BookieSocketAddress bookieSrc, final BookieSocketAddress bookieDest,
+    public void asyncRecoverBookieData(final BookieSocketAddress bookieSrc,
```

Comment:
this change is anyway a semantic breaking change. the original semantic is wrong and doesn't respect placement policy. so we have to break BC.

Reply:
Well if the operator knows what they are doing it can be performed by respecting the placement policy. But I agree with the change as it is good to be fungible and let the engine do the right thing.